scipy/stats/_distn_infrastructure.py

```diff
@@ -2143,8 +2143,17 @@ def _reduce_func(self, args, kwds):
             else:
                 x0.append(args[n])
 
+        objectives = {"MLE": self._penalized_nnlf,
+                      "GMM": self._moment_error_norm}
+        method = kwds.pop('method', "mle").upper()
```

Comment:
I'm not sure if we have an agreed convention here but everything I've seen usually uses lowercasings, a few examples: 
https://github.com/scipy/scipy/blob/c1372d8aa90a73d8a52f135529293ff4edb98fc8/scipy/stats/mstats_basic.py#L1295
https://github.com/scipy/scipy/blob/c1372d8aa90a73d8a52f135529293ff4edb98fc8/scipy/io/matlab/byteordercodes.py#L55
https://github.com/scipy/scipy/blob/c1372d8aa90a73d8a52f135529293ff4edb98fc8/scipy/constants/constants.pyv
https://github.com/scipy/scipy/blob/c1372d8aa90a73d8a52f135529293ff4edb98fc8/scipy/io/arff/arffread.py#L248
so is there a strong reason to not use `lower`?

Reply:
I know that's the usual convention, but I did this in case the user enters an invalid `method`. I wanted the error message to report the valid options:
```
        except KeyError:
            raise ValueError("Method '{0}' not available; must be one of {1}"
                             .format(method, str(set(objectives.keys()))))
```
and I wanted them to be capitalized in that case. Clearly I can still accomplish my objectives  with lower-case keys in the dictionary; it just would take a teeny bit more code.

Reply:
>Clearly I can still accomplish my objectives with lower-case keys in the dictionary; it just would take a teeny bit more code.

Is there a reason why you couldn't simply change the keys in the `objectives` dictionary to lower case?

Reply:
I wanted them to appear capitalized in the error message:
```Method FOO not available; must be one of {MLE, MM}```
But for consistency, I changed this to lower while maintaining the uppercase capitalization in the error message. The code is just a little less streamlined.

Reply:
>But for consistency, I changed this to lower while maintaining the uppercase capitalization in the error message. The code is just a little less streamlined.

Thanks.